jupyter_book/commands/__init__.py

```diff
@@ -68,27 +71,22 @@ def build(path_book, path_output, config, toc, warningiserror, builder):
     book_config["globaltoc_path"] = str(toc)
 
     # Configuration file
-    if config is None:
+    path_config = config
+    if path_config is None:
+        # Check if there's a `_config.yml` file in the source directory
         if PATH_BOOK.joinpath("_config.yml").exists():
-            config = PATH_BOOK.joinpath("_config.yml")
-
-    extra_extensions = None
-    if config is not None:
-        book_config["yaml_config_path"] = str(config)
-        config_yaml = yaml.safe_load(config.read_text())
-        # Pop the extra extensions since we need to append, not replace
-        extra_extensions = config_yaml.pop("sphinx", {}).get("extra_extensions")
-        # Support Top Level config Passthrough
-        # https://www.sphinx-doc.org/en/latest/usage/configuration.html#project-information
-        sphinx_options = ["project", "author", "copyright"]
-        for option in sphinx_options:
-            if option in config_yaml.keys():
-                book_config[option] = config_yaml[option]
+            path_config = str(PATH_BOOK.joinpath("_config.yml"))
+
+    if path_config:
+        if not Path(path_config).exists():
+            raise ValueError(f"Config file path given, but not found: {path_config}")
+        config_yaml = yaml.safe_load(Path(path_config).read_text())
```

Comment:
Unless I've missed something, it seems like the only reason `_config.yml` (or whatever book-specific configuration is supplied by the user via the `--config` flag) is read is to build the `latex_config`. I haven't traced this all the way through, but might this result in ordering issues for the latex-specific config vals (since both `path_config`, and `latex_config` are passed into `build_sphinx`)? I guess it's necessary because the `--builder` flag dictates whether the latex config values are relevant.

Reply:
lemme think about that a bit, I wonder if there is a way to handle the latex logic within the `build_sphinx` function...

also, can you try re-creating the bug you had originally opened up with the code in this PR? (the one with `page` and `--execute` I believe)

Reply:
ok the latest commit moves all the latex builder logic inside the `build_sphinx` function. It is a little bit unclear to me exactly what it's doing, but i *think* that it should be have the exact same way haha, let's see what the tests do

Reply:
The original bug pointed out in the discussion in #594 is not present in this PR - the CLI override for page execution is respected without having to touch the rest of the configuration machinery :)

Reply:
💥